fgspectra/frequency.py

```diff
@@ -3,120 +3,142 @@
 Frequency-dependent foreground components.
 
 This module implements the frequency-dependent component of common foreground
-contaminants. This consist only of factorizable components -- i.e. the components
-in this module describe :math:`f(\nu)` for which the component X has spectrum
-
-.. math:: C_{\ell}^{X} = f(\nu) C^X_{0,\ell}
+contaminants. 
 
 This package draws inspiration from FGBuster (Davide Poletti and Josquin Errard)
 and BeFoRe (David Alonso and Ben Thorne).
 """
+
+from abc import ABC, abstractmethod
 import numpy as np
 from scipy import constants
 
 # TODO: we need to figure out the unit situation.
 
-class FrequencySpectrum:
+T_CMB = 2.725
+
+
+class FrequencySpectrum(ABC):
```

Comment:
What about naming it `SED`? It would be great if we use the word "spectrum" only for power spectra